tensorrt_llm/_torch/pyexecutor/_util.py

```diff
@@ -586,15 +586,14 @@ def instantiate_sampler(engine: PyTorchModelEngine,
     if engine.spec_config is not None and engine.spec_config.spec_dec_mode.has_spec_decoder(
     ):
         return get_spec_decoder(sampler_args, engine.spec_config)
-    if pytorch_backend_config.enable_trtllm_sampler:
-        decoding_mode = get_decoding_mode(executor_config)
-        return TRTLLMSampler(executor_config, engine.model, engine.dtype,
-                             mapping, decoding_mode,
-                             pytorch_backend_config.disable_overlap_scheduler)
+    if pytorch_backend_config.enable_torch_sampler or pytorch_backend_config.enable_mixed_sampler or engine.spec_config is not None:
```

Comment:
Can `engine.spec_config is not None` be True while `engine.spec_config.spec_dec_mode.has_spec_decoder(
    )` is False? I think in the previous `if` on L586, `engine.spec_config is not None` is just a type guard, equivalent to optional chaining. Or am I missing something?